include/seqan3/range/container/concatenated_sequences.hpp

```diff
@@ -0,0 +1,1280 @@
+    //!\hideinitializer
+    using const_reference = decltype(std::as_const(data_values) | ranges::view::slice(0, 1) | ranges::view::const_);
+
+    //!\brief The iterator type of this container (a random access iterator).
+    //!\hideinitializer
+    using iterator = detail::random_access_iterator<concatenated_sequences>;
+
+    //!\brief The const iterator type of this container (a random access iterator).
+    //!\hideinitializer
+    using const_iterator = detail::random_access_iterator<concatenated_sequences const>;
+
+    //!\brief A signed integer type (usually std::ptrdiff_t)
+    //!\hideinitializer
+    using difference_type = std::ptrdiff_t;
+
+    //!\brief An unsigned integer type (usually std::size_t)
+    //!\hideinitializer
+    using size_type = size_t;
+    //!\}
+
+    /* rule of six */
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief Default constructors.
+    concatenated_sequences() = default;
+    //!\brief Default constructors.
+    constexpr concatenated_sequences(concatenated_sequences const &) = default;
+    //!\brief Default constructors.
+    constexpr concatenated_sequences(concatenated_sequences &&) = default;
+    //!\brief Default constructors.
+    constexpr concatenated_sequences & operator=(concatenated_sequences const &) = default;
+    //!\brief Default constructors.
+    constexpr concatenated_sequences & operator=(concatenated_sequences &&) = default;
+    //!\brief Default constructors.
+    ~concatenated_sequences() = default;
+
+    /*!\brief Construct/assign from a different range.
+     * \tparam rng_of_rng_type The type of range to be inserted; must be "like this", i.e. satisfy
+     * seqan3::input_range_concept and its `value_type` mus be like `value_type` (also
+     * seqan3::input_range_concept and same `value_type`s.
+     * \param rng_of_rng The sequences to construct/assign from.
+     */
+    template <typename rng_of_rng_type>
+    concatenated_sequences(rng_of_rng_type && rng_of_rng)
+        requires input_range_concept<std::decay_t<rng_of_rng_type>> /*&&
+                 like_value_type<std::decay_t<ranges::value_type_t<std::decay_t<rng_of_rng_type>>>>*/
+    {
+        if constexpr (sized_range_concept<std::decay_t<rng_of_rng_type>>)
+            data_delimiters.reserve(ranges::size(rng_of_rng) + 1);
+
+        for (auto && val : rng_of_rng)
+        {
+            data_values.insert(data_values.end(), val.begin(), val.end());
+            data_delimiters.push_back(data_delimiters.back() + val.size());
+        }
+    }
+
+    /*!\brief Construct/assign with `count` times `value`.
+     * \tparam rng_type The type of range to be inserted; must be "like" `value_type`, i.e. satisfy
+     * seqan3::input_range_concept and have the same `value_type` as `value_type`.
+     * \param count Number of elements.
+     * \param value The initial value to be assigned.
+     *
+     * \par Complexity
+     *
+     * In \f$O(count)\f$.
```

Comment:
ähmm, you mean `count * size(value)`? I assume, in line 219 you probably copy the values into the concatenator?